pkg/reconciler/v1alpha1/pipelinerun/pipelinerun_test.go

```diff
@@ -1005,3 +1307,225 @@ func TestReconcilePropagateAnnotations(t *testing.T) {
+	condCheck1 := clients.Pipeline.Actions()[1].(ktesting.CreateAction).GetObject().(*v1alpha1.TaskRun)
+	if condCheck0 == nil || condCheck1 == nil {
+		t.Errorf("Expected two ConditionCheck TaskRuns to be created, but it wasn't.")
+	}
+
+	actual := []*v1alpha1.TaskRun{condCheck0, condCheck1}
+	if d := cmp.Diff(actual, expectedConditionChecks); d != "" {
+		t.Errorf("expected to see 2 ConditionCheck TaskRuns created. Diff %s", d)
+	}
+}
+
+func TestReconcileWithFailingConditionChecks(t *testing.T) {
+	names.TestingSeed()
+	conditions := []*v1alpha1.Condition{{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "always-false",
+			Namespace: "foo",
+		},
+		Spec: v1alpha1.ConditionSpec{
+			Check: corev1.Container{
+				Image: "foo",
+				Args:  []string{"bar"},
+			},
+		},
+	}}
+	pipelineRunName := "test-pipeline-run-with-conditions"
+	prccs := make(map[string]*v1alpha1.PipelineRunConditionCheckStatus)
+
+	conditionCheckName := pipelineRunName + "task-2-always-false-xxxyyy"
+	prccs[conditionCheckName] = &v1alpha1.PipelineRunConditionCheckStatus{
+		ConditionName: "always-false",
+		Status:        &v1alpha1.ConditionCheckStatus{},
+	}
+	ps := []*v1alpha1.Pipeline{tb.Pipeline("test-pipeline", "foo", tb.PipelineSpec(
+		tb.PipelineTask("task-1", "hello-world"),
+		tb.PipelineTask("task-2", "hello-world", tb.PipelineTaskCondition("always-false")),
+		tb.PipelineTask("task-3", "hello-world", tb.RunAfter("task-1")),
+	))}
+
+	prs := []*v1alpha1.PipelineRun{tb.PipelineRun("test-pipeline-run-with-conditions", "foo",
+		tb.PipelineRunAnnotation("PipelineRunAnnotation", "PipelineRunValue"),
+		tb.PipelineRunSpec("test-pipeline",
+			tb.PipelineRunServiceAccount("test-sa"),
+		),
+		tb.PipelineRunStatus(tb.PipelineRunStatusCondition(apis.Condition{
+			Type:    apis.ConditionSucceeded,
+			Status:  corev1.ConditionUnknown,
+			Reason:  resources.ReasonRunning,
+			Message: "Not all Tasks in the Pipeline have finished executing",
+		}), tb.PipelineRunTaskRunsStatus(pipelineRunName+"task-1", &v1alpha1.PipelineRunTaskRunStatus{
+			PipelineTaskName: "task-1",
+			Status:           &v1alpha1.TaskRunStatus{},
+		}), tb.PipelineRunTaskRunsStatus(pipelineRunName+"task-2", &v1alpha1.PipelineRunTaskRunStatus{
+			PipelineTaskName: "task-2",
+			Status:           &v1alpha1.TaskRunStatus{},
+			ConditionChecks:  prccs,
+		})),
+	)}
+
+	ts := []*v1alpha1.Task{tb.Task("hello-world", "foo")}
+	trs := []*v1alpha1.TaskRun{
+		tb.TaskRun(pipelineRunName+"task-1", "foo",
+			tb.TaskRunOwnerReference("kind", "name"),
+			tb.TaskRunLabel(pipeline.GroupName+pipeline.PipelineLabelKey, "test-pipeline-run-with-conditions"),
+			tb.TaskRunLabel(pipeline.GroupName+pipeline.PipelineRunLabelKey, "test-pipeline"),
+			tb.TaskRunSpec(tb.TaskRunTaskRef("hello-world")),
+			tb.TaskRunStatus(tb.Condition(apis.Condition{
+				Type:   apis.ConditionSucceeded,
+				Status: corev1.ConditionTrue,
+			}),
+			),
+		),
+		// TODO: add a test builder for conditions/checks
```

Comment:
hee hee my apologies for asking for this in a comment XD 🙏 

If you want to leave it as a TODO, i like using the google type syntax and indicating what issue it'll be fixed as part of, like:

```
// TODO(#12345)
```